src/odemis/driver/keysight.py

```diff
@@ -0,0 +1,496 @@
+        logging.debug("Sending query %s", to_str_escape(q))
+        response = self._accesser.sendQuery(q)
+
+        return response
+
+    def setOutput(self, c, p: bool):
+        if p:
+            self._sendCommand(b"OUTP%d ON" % c)
+        else:
+            self._sendCommand(b"OUTP%d OFF" % c)
+
+    def setFrequency(self, c, f):
+        """
+        c (int, 1 or 2): the channel to set the frequency of.
+        f (float): the frequency value in Hertz
+        """
+        self._sendCommand(b"sour%d:freq %e" %(c, f))
+
+    def setVoltageMin(self, c, v):
+        """
+        c (int): the channel to set the low voltage. 1 or 2
+        v (float): the low voltage value
+        """
+        self._sendCommand(b"sour%d:volt:low %f" % (c, v))
+
+    def setVoltageMax(self, c, v):
+        """
+        c (int, 1 or 2): the channel to set the high voltage of.
+        v (float): the high voltage value
+        """
+        self._sendCommand(b"sour%d:volt:high %f" % (c, v))
+
+    def setVoltageLimitMin(self, c, v):
+        """
+        c (int, 1 or 2): the channel to set the minimum voltage limit of.
+        v (float): the minimum voltage limit
+        """
+        self._sendCommand(b"sour%d:volt:lim:low %f" % (c, v))
+
+    def setVoltageLimitMax(self, c, v):
+        """
+        c (int, 1 or 2): the channel to set the maximum voltage limit of.
+        v (float): the maximum voltage limit
+        """
+        self._sendCommand(b"sour%d:volt:lim:high %f" % (c, v))
+
+    def setTriggerDelay(self, c, d):
+        """
+        c (int, 1 or 2): the channel to set the delay of.
+        d (float): the trigger delay to set. between 0 and 1000 s
+        """
+        self._sendCommand(b"trig%d:del %e" % (c, d))
+
+    def setPhase(self, c: int, t: float):
+        """
+        c (1 or 2): the channel to set the delay of.
+        t: phase value in time (s). Can only be between -period and +period
+        """
+        self._sendCommand(b"sour%d:phas %e sec" % (c, t))
+        # self._sendCommand(b"phas 1e-6 SEC")  #DEBUG
+
+    def setDutyCycle(self, c, dc):
+        """
+        c (int, 1 or 2): the channel to set the duty cycle of.
+        dc (float): the duty cycle percentage. between 00.01 and 99.99
+        """
+        self._sendCommand(b"sour%d:func:squ:dcyc %.2f" % (c, dc))
+
+    def setTracking(self, c: int, t: str) -> None:
+        """
+        c (1 or 2): the channel to set the tracking mode of.
+        t (ON, OFF or INV): the tracking mode
+        """
+        if t not in ["ON", "OFF", "INV"]:
+            raise ValueError("Incorrect tracking mode: %s given" % (t,))
+        self._sendCommand(b"sour%d:trac %s" % (c, t.encode("ascii")))
+
+    def setWaveform(self, c, f):
+        """
+        c (int, 1 or 2): the channel to set the tracking mode of.
+        f (str SIN, SQU, RAMP or PULS): the wave form to generate
+        """
+        self._sendCommand(b"sour%d:appl:%s" % (c, f))
```

Comment:
Here would be good to check for the input value, to check it's only within the acceptable values.